scenarioo-server/src/main/java/org/scenarioo/business/diffViewer/ComparisonExecutor.java

```diff
@@ -170,7 +253,7 @@ private BuildDiffInfo saveBuildDiffInfoWithStatus(ComparisonParameters compariso
 		return saveBuildDiffInfo(comparisonParameters, buildDiffInfo);
 	}
 
-	private BuildDiffInfo saveBuildDiffInfo(ComparisonParameters comparisonParameters, BuildDiffInfo buildDiffInfo) {
+	private synchronized BuildDiffInfo saveBuildDiffInfo(ComparisonParameters comparisonParameters, BuildDiffInfo buildDiffInfo) {
 
 		if (comparisonParameters == null) {
```

Comment:
I think we don't necessarily need these two null checks. On the other hand they might make problem analysis a bit easier. I would replace them with shorter statements:

```
		Preconditions.checkNotNull(comparisonParameters,
			"comparisonParameters must not be null");
		Preconditions.checkNotNull(comparisonParameters.getComparisonConfiguration(),
			"comparisonParameters.getComparisonConfiguration() must not be null");
```

Reply:
hmm, I think this is not important right now - better be safe for easier problem analysis.

Reply:
@adiherzog you can further improve this if you like too - but to be honest I prefer to not touch it too much anymore - unless there is a bug 😉